client/state/user-settings/reducer.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge, omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	USER_SETTINGS_FETCH,
+	USER_SETTINGS_FETCH_SUCCESS,
+	USER_SETTINGS_FETCH_FAILURE,
+	USER_SETTINGS_SAVE_SUCCESS,
+	USER_SETTINGS_RECEIVE,
+	USER_SETTINGS_UNSAVED_SET,
+	USER_SETTINGS_UNSAVED_REMOVE,
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+import { decodeEntities } from 'lib/formatting';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function decodeUserSettingsEntities( data ) {
+	const decodedValues = {
+		display_name: data.display_name && decodeEntities( data.display_name ),
+		description: data.description && decodeEntities( data.description ),
+		user_URL: data.user_URL && decodeEntities( data.user_URL )
+	};
+
+	return merge( {}, data, decodedValues );
+}
+
+/**
+ * Returns the updated remote values state after an action has been dispatched.
+ * The remote values state reflects preferences which are persisted to the REST
+ * API current user settings endpoint.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const settings = createReducer( false, {
+	[ USER_SETTINGS_RECEIVE ]: ( state, { values } ) => {
+		return merge( {}, state || {}, decodeUserSettingsEntities( values ) );
+	}
+} );
```

Comment:
> I don't see much value gained in using `createReducer` here since we have a very simple reducer which doesn't handle serialization or memoization. Is there a reason I'm missing why it's used?

I use `createReducer` because it provides a nice API where I pass a map `actionType -> handler` instead of creating a big `switch` statement. And also because it seems to be used everywhere else. Or is the preferred style to use the good old `switch`?

The default implementations for `SERIALIZE` and `DESERIALIZE` return the initial state. So, `DESERIALIZE` does the state hydration, and `SERIALIZE` kind of annihilates the state (?). I don't quite understand what these actions are supposed to do, except schema validation.

I'm not sure what "memoization" means when talking about a reducer.